lib/elixir/src/elixir_expand.erl

```diff
@@ -1020,4 +1039,8 @@ format_error({underscored_var_access, Name, Kind}) ->
   io_lib:format("the underscored variable \"~ts\"~ts is used after being set. "
                 "A leading underscore indicates that the value of the variable "
                 "should be ignored. If this is intended please rename the "
-                "variable to remove the underscore", [Name, context_info(Kind)]).
+                "variable to remove the underscore", [Name, context_info(Kind)]);
+format_error(struct_comparison) ->
+  "comparison operators (>, <, >=, <=) perform structural and not semantic comparison, "
+  "comparing with a struct literal is unlikely to give a meaningful result. "
+  "Many modules export a compare/2 or cmp/2 function that can be used for semantic comparison".
```

Comment:
I don't feel like we should talk about `cmp/2` since we have settled on `compare/2`. It is also nice to give the name of the struct and start the error message with a small sentence. Here is my suggestion:

> invalid comparison with struct literal ~ts. Comparison operators (>, <, >=, <=) perform structural
> and not semantic comparison. Comparing with a struct literal is unlikely to give a meaningful result.
> Modules typically define a compare/2 function that can be used for semantic comparison.